src/vs/editor/contrib/stickyScroll/browser/stickyScrollProvider.ts

```diff
@@ -0,0 +1,177 @@
+
+	static readonly ID = 'store.contrib.stickyScrollController';
+	private readonly editor: ICodeEditor;
+	private readonly languageFeaturesService: ILanguageFeaturesService;
+	private readonly updateSoon: RunOnceScheduler;
+
+	private cts: CancellationTokenSource | undefined;
+	private outlineModel: StickyOutlineElement | undefined;
+	private readonly sessionStore: DisposableStore = new DisposableStore();
+	private modelVersionId: number = 0;
+
+	constructor(
+		editor: ICodeEditor,
+		@ILanguageFeaturesService _languageFeaturesService: ILanguageFeaturesService,
+	) {
+		super();
+		this.editor = editor;
+		this.languageFeaturesService = _languageFeaturesService;
+		this.updateSoon = this._register(new RunOnceScheduler(() => this.update(true), 50));
+		this._register(this.editor.onDidChangeConfiguration(e => {
+			if (e.hasChanged(EditorOption.experimental)) {
+				this.readConfiguration();
+			}
+		}));
+		this.readConfiguration();
+	}
+
+	private readConfiguration() {
+		const options = this.editor.getOption(EditorOption.experimental);
+		if (options.stickyScroll.enabled === false) {
+			this.sessionStore.clear();
+			return;
+		} else {
+			this.sessionStore.add(this.editor.onDidChangeModel(() => this.update(true)));
+			this.sessionStore.add(this.editor.onDidScrollChange(() => this.update(false)));
+			this.sessionStore.add(this.editor.onDidChangeHiddenAreas(() => this.update(true)));
+			this.sessionStore.add(this.editor.onDidChangeModelContent(() => this.updateSoon.schedule()));
+			this.sessionStore.add(this.languageFeaturesService.documentSymbolProvider.onDidChange(() => this.update(true)));
+			const lineNumberOption = this.editor.getOption(EditorOption.lineNumbers);
+			if (lineNumberOption.renderType === RenderLineNumbersType.Relative) {
+				this.sessionStore.add(this.editor.onDidChangeCursorPosition(() => this.update(false)));
+			}
+			this.update(true);
+		}
+	}
+
+	public getVersionId() {
+		return this.modelVersionId;
+	}
+
+	private async update(updateOutline: boolean = false): Promise<void> {
+		if (updateOutline) {
+			this.cts?.dispose(true);
+			this.cts = new CancellationTokenSource();
+			await this.updateOutlineModel(this.cts.token);
+		}
+		this.onStickyScrollChangeEmitter.fire();
+	}
+
+	private async updateOutlineModel(token: CancellationToken) {
+		if (this.editor.hasModel()) {
+			const model = this.editor.getModel();
+			const modelVersionId = model.getVersionId();
+			const outlineModel = await OutlineModel.create(this.languageFeaturesService.documentSymbolProvider, model, token) as OutlineModel;
+			if (token.isCancellationRequested) {
+				return;
+			}
+			this.outlineModel = StickyOutlineElement.fromOutlineModel(outlineModel);
+			this.modelVersionId = modelVersionId;
+		}
+	}
+
+	public getCandidateStickyLinesIntersectingFromOutline(range: StickyRange, outlineModel: StickyOutlineElement, stickyLineCandidates: StickyLineCandidate[], depth: number, lastStartLineNumber: number): void {
```

Comment:
I would call `stickyLineCandidates`  `results`, to make sure it is an output.
The type already makes it clear what is in that array.